app/controllers/HomeController.scala

```diff
@@ -34,18 +34,25 @@ object HomeController extends HomeController(ApplicationClient)
 class HomeController(applicationClient: ApplicationClient) extends BaseController(applicationClient) {
   val Withdrawer = "Candidate"
 
+
   val present = CSRSecureAction(ActiveUserRole) { implicit request => implicit cachedData =>
     cachedData.application.map { application =>
+
+
+      def getPhase2Test: Future[Option[Phase2TestGroupWithNames]] = if (application.applicationStatus == ApplicationStatus.PHASE2_TESTS) {
+        applicationClient.getPhase2TestProfile(application.applicationId).map(Some(_))
```

Comment:
getPhase2TestProfile should always return something - we shouldn't be calling it if we're not in a state to expect the test to have been created.  I'm just wrapping the result in a Some to handle cases where we don't want to actually call it.